Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,535 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import { documentSelector } from '../common';
+import { Client } from './client';
+import { ReferencesCommandMode, referencesCommandModeToString } from './references';
+import { getCustomConfigProviders, CustomConfigurationProviderCollection, isSameProviderExtensionId } from './customProviders';
+import * as nls from 'vscode-nls';
+import { setTimeout } from 'timers';
+import { CppSettings } from './settings';
+import { UI } from './ui';
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+interface IndexableQuickPickItem extends vscode.QuickPickItem {
+    index: number;
+}
+interface KeyedQuickPickItem extends vscode.QuickPickItem {
+    key: string;
+}
+
+// Higher numbers mean greater priority.
+enum ConfigurationPriority {
+    IncludePath = 1,
+    CompileCommands = 2,
+    CustomProvider = 3,
+}
+
+interface ConfigurationStatus {
+    configured: boolean;
+    priority: ConfigurationPriority;
+}
+
+enum LanguageStatusPriority {
+    First = 0,
+    High = 1,
+    Mid = 2,
+    Low = 3
+}
+
+export class NewUI implements UI {
+    private configStatusBarItem: vscode.LanguageStatusItem;
+    private browseEngineStatusBarItem: vscode.LanguageStatusItem;
+    private intelliSenseStatusBarItem: vscode.LanguageStatusItem;
+    private referencesStatusBarItem: vscode.StatusBarItem;
+    private codeAnalysisStatusBarItem: vscode.LanguageStatusItem;
+    private configDocumentSelector: vscode.DocumentFilter[] = [
+        { scheme: 'file', language: 'c' },
+        { scheme: 'file', language: 'cpp' },
+        { scheme: 'file', language: 'cuda-cpp' },
+        { scheme: 'file', language: 'jsonc', pattern: '**/.vscode/*.json'},
```

Comment:
The .vscode/c_cpp_properties.json isn't getting picked up -- I think you need to add `        { scheme: 'file', language: 'json', pattern: '**/.vscode/*.json'},`. Or maybe there's some bug with us not associating that file with jsconc....